ensemble/ensemble.go

```diff
@@ -0,0 +1,184 @@
+		}
+	}
+
+	// Microcks container
+	if ensemble.postmanEnabled {
+		ensemble.microcksContainerOptions.Add(microcks.WithEnv("POSTMAN_RUNNER_URL", "http://postman:3000"))
+	}
+	ensemble.microcksContainerOptions.Add(microcks.WithEnv("TEST_CALLBACK_URL", "http://microcks:8080"))
+	ensemble.microcksContainer, err = microcks.RunContainer(ctx, ensemble.microcksContainerOptions.list...)
+	if err != nil {
+		return nil, err
+	}
+
+	// Postman container
+	if ensemble.postmanEnabled {
+		ensemble.postmanContainer, err = postman.RunContainer(ctx, ensemble.postmanContainerOptions.list...)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return ensemble, nil
+}
+
+// WithDefaultNetwork allows to use a default network
+func WithDefaultNetwork() Option {
+	return func(e *MicrocksContainersEnsemble) (err error) {
+		e.network, err = network.New(e.ctx, network.WithCheckDuplicate())
+		if err != nil {
+			return err
+		}
+
+		e.microcksContainerOptions.Add(microcks.WithNetwork(e.network.Name))
+		e.microcksContainerOptions.Add(microcks.WithNetworkAlias(e.network.Name, defaultNetworkAliasMicrocks))
+		e.postmanContainerOptions.Add(postman.WithNetwork(e.network.Name))
+		e.postmanContainerOptions.Add(postman.WithNetworkAlias(e.network.Name, defaultNetworkAliasPostman))
+
+		return nil
+	}
+}
+
+// WithNetwork allows to define the network
+func WithNetwork(network *testcontainers.DockerNetwork) Option {
+	return func(e *MicrocksContainersEnsemble) error {
+		e.network = network
+		e.microcksContainerOptions.Add(microcks.WithNetwork(e.network.Name))
+		e.microcksContainerOptions.Add(microcks.WithNetworkAlias(e.network.Name, defaultNetworkAliasMicrocks))
+		e.postmanContainerOptions.Add(postman.WithNetwork(e.network.Name))
+		e.postmanContainerOptions.Add(postman.WithNetworkAlias(e.network.Name, defaultNetworkAliasPostman))
+		return nil
+	}
+}
+
+// WithMainArtifact provides paths to artifacts that will be imported as main or main
+// ones within the Microcks container.
+// Once it will be started and healthy.
+func WithMainArtifact(artifactFilePath string) Option {
+	return func(e *MicrocksContainersEnsemble) error {
+		e.microcksContainerOptions.Add(microcks.WithMainArtifact(artifactFilePath))
+		return nil
+	}
+}
+
+// WithSecondaryArtifact provides paths to artifacts that will be imported as main or main
+// ones within the Microcks container.
+// Once it will be started and healthy.
+func WithSecondaryArtifact(artifactFilePath string) Option {
+	return func(e *MicrocksContainersEnsemble) error {
+		e.microcksContainerOptions.Add(microcks.WithSecondaryArtifact(artifactFilePath))
+		return nil
+	}
+}
+
+// WithPostman allows to enable Postman container
+func WithPostman(enable bool) Option {
+	return func(e *MicrocksContainersEnsemble) error {
+		e.postmanEnabled = enable
+		return nil
+	}
+}
+
+// WithPostmanImage helps to use specific Postman image
+func WithPostmanImage(image string) Option {
+	return func(e *MicrocksContainersEnsemble) error {
```

Comment:
Why not enforcing `e.postmanEnabled = true` if this function is invoked?

Reply:
Yes, I'll do! Very good suggestion!